### DISCUSSION TITLE
How to extend Client and Socket types in Socket.io@4.2.0 and typescript@4.4.3

### DISCUSSION THREAD

**Discussion by ehsanshekari** [socketio/socket.io#4122] [Q&A] [answered]
I have a Node.js Server with Typescript and now I want to update the socket.io depencency to 4.2.0.
I need to extend Client and Socket objects. Previously, in socket.io@2.x.x I could do something like below:

```
export interface CustomClient extends Client {
  customField: string;
}

export interface CustomSocket extends Socket {
  client: CustomClient;
}
```

How can I achieve the same thing with socket.io@4.2.0?

**Comment by panwauu** [comment, marked as answer]
On Socket there is a data field which is any typed. So you can put any additional data there
